src/nodes/typeGuards/compound.ts

```diff
@@ -61,6 +61,12 @@ export function isJsxTagNamePropertyAccess(
 	);
 }
 
+export function isNamedDeclaration(
+	node: ts.Declaration
+): node is ts.NamedDeclaration {
+	return "name" in node;
+}
```

Comment:
🤔 I'm not 100% that this is right... but couldn't think of anything better given how small the `ts.NamedDeclaration` type is:

```ts
export interface NamedDeclaration extends Declaration {
    readonly name?: DeclarationName;
}
```

Reply:
Probably best to add an `&& isDeclarationName(node.name)`

Reply:
Ah good point. `node.name` is optional, so I guess this is needed only if it exists?

Reply:
Actually, it being optional is a bit confusing. Surely if it doesn't have a name it isn't a named declaration.

I think for now it might be best to just go with:

 ```ts
const isNamedDeclaration = isDeclaration as (node: TS.Node) => node is ts NamedDeclaration
```

Seems the name prop doesn't need to exist at all.

Reply:
Yeah I'll just go with the latter for now.